pkg/allowedpaths/cmd.go

```diff
@@ -0,0 +1,784 @@
+	}
+	return newCmd(ctx, fullPathToCmdValidated, arg...), nil
+}
+
+func Xrdb(ctx context.Context, arg ...string) (*exec.Cmd, error) {
+	if runtime.GOOS != "linux" {
+		return nil, errors.New("xrdb supported on linux only")
+	}
+
+	fullPathToCmdValidated, err := validatedPath("/usr/bin/xrdb")
+	if err != nil {
+		return nil, fmt.Errorf("xrdb not found: %w", err)
+	}
+	return newCmd(ctx, fullPathToCmdValidated, arg...), nil
+}
+
+func Xwwwbrowser(ctx context.Context, arg ...string) (*exec.Cmd, error) {
+	if runtime.GOOS != "linux" {
+		return nil, errors.New("x-www-browser supported on linux only")
+	}
+
+	fullPathToCmdValidated, err := validatedPath("/usr/bin/x-www-browser")
+	if err != nil {
+		return nil, fmt.Errorf("x-www-browser not found: %w", err)
+	}
+	return newCmd(ctx, fullPathToCmdValidated, arg...), nil
+}
+
+func Zerotiercli(ctx context.Context, arg ...string) (*exec.Cmd, error) {
+	var fullPathToCmd string
+	switch runtime.GOOS {
+	case "darwin", "linux":
+		fullPathToCmd = "/usr/local/bin/zerotier-cli"
+	case "windows":
+		fullPathToCmd = path.Join(os.Getenv("SYSTEMROOT"), "ProgramData", "ZeroTier", "One", "zerotier-one_x64.exe")
+	}
+
+	fullPathToCmdValidated, err := validatedPath(fullPathToCmd)
+	if err != nil {
+		return nil, fmt.Errorf("zerotier-cli not found: %w", err)
+	}
+
+	// For windows, "-q" should be prepended before all other args
+	if runtime.GOOS == "windows" {
+		arg = append([]string{"-q"}, arg...)
+	}
+
+	return newCmd(ctx, fullPathToCmdValidated, arg...), nil
+}
+
+func Zfs(ctx context.Context, arg ...string) (*exec.Cmd, error) {
+	if runtime.GOOS != "linux" {
+		return nil, errors.New("zfs supported on linux only")
+	}
+
+	fullPathToCmdValidated, err := validatedPath("/usr/sbin/zfs")
+	if err != nil {
+		return nil, fmt.Errorf("zfs not found: %w", err)
+	}
+	return newCmd(ctx, fullPathToCmdValidated, arg...), nil
+}
+
+func Zpool(ctx context.Context, arg ...string) (*exec.Cmd, error) {
+	if runtime.GOOS != "linux" {
+		return nil, errors.New("zpool supported on linux only")
+	}
+
+	fullPathToCmdValidated, err := validatedPath("/usr/sbin/zpool")
+	if err != nil {
+		return nil, fmt.Errorf("zpool not found: %w", err)
+	}
+	return newCmd(ctx, fullPathToCmdValidated, arg...), nil
+}
+
+func newCmd(ctx context.Context, fullPathToCmd string, arg ...string) *exec.Cmd {
+	return exec.CommandContext(ctx, fullPathToCmd, arg...) //nolint:forbidigo
+}
+
+func validatedPath(knownPath string) (string, error) {
+	knownPath = filepath.Clean(knownPath)
+
+	if _, err := os.Stat(knownPath); err == nil {
+		return knownPath, nil
+	}
+
+	// Not found at known location -- return error for darwin and windows.
+	// We expect to know the exact location for allowlisted commands on all
+	// OSes except for a few Linux distros.
+	if runtime.GOOS != "linux" {
+		return "", fmt.Errorf("not found: %s", knownPath)
+	}
+
+	cmdName := filepath.Base(knownPath)
+	if foundPath, err := exec.LookPath(cmdName); err == nil {
+		return foundPath, nil
+	}
```

Comment:
I don't think we want this for all linuxes

Reply:
Updated!